templates/vue/src/components/FileUpload.vue

```diff
@@ -0,0 +1,135 @@
+<template>
+  <b-container fluid>
+    <b-row>
+        <b-col>
+            <b-form-file
+              ref="file"
+              name="async-upload"
+              class="image-file mb-2"
+              placeholder="Choose a file or drop it here..."
+              drop-placeholder="Drop file here..."
+              @change="uploadFile"
+              required>
+            </b-form-file>
+        </b-col>
+        <b-col sm="1" class="divider">
+          <h6>OR</h6>
+        </b-col>
+        <b-col>
+          <b-row>
+            <b-form-input
+              :id="inputId"
+              :placeholder="placeholderText"
+              :value="value"
+              @input="$emit('input', $event)"
+              required
+            />
+          </b-row>
+        </b-col>
+    </b-row>
+    <b-row>
+      <progress max="100" :value.prop="uploadPercentage"></progress>
+    </b-row>
+  </b-container>
+</template>
+
+<script>
+import axios from "axios"
+
+export default {
+  name: "file-upload",
+  props: {
+    inputId: {
+      type: String,
+      required: true,
+    },
+    placeholderText: {
+      type: String,
+      required: true,
+    },
+    value: {
+      type: String,
+      required: false
+    }
+  },
+
+  data() {
+    return{
+      uploadPercentage: 0
+    }
+  },
+
+  methods: {
+    uploadFile(event) {
+      var formData = new FormData()
+      formData.append('action', 'upload-attachment')
+      formData.append('async-upload', event.target.files[0])
+      formData.append('name', event.target.files[0].name)
+      formData.append('_wpnonce', wpData.file_upload_nonce)
+
+      axios
+        .post( wpData.upload_url, formData,
+          {
+            headers: {
+              'Content-Type': 'multipart/form-data'
+            },
+            onUploadProgress: function( progressEvent ) {
+              this.uploadPercentage = parseInt( Math.round( ( progressEvent.loaded / progressEvent.total ) * 100 ))
+            }.bind(this)
+          }
+        )
+        .then(response => {
+          this.$emit('input', response.data.data.url)
+        })
```

Comment:
Let's keep the code styling consistent. Make sure the linter runs before you commit. If it doesn't, let me know and we can figure out why it's not running automatically.